sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -756,10 +756,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       val invalidCols = groupByCols -- correlatedCols
       // GROUP BY columns must be a subset of columns in the predicates
       if (invalidCols.nonEmpty) {
-        failAnalysis(
-          "A GROUP BY clause in a scalar correlated subquery " +
-            "cannot contain non-correlated columns: " +
-            invalidCols.mkString(","))
+        throw QueryCompilationErrors.nonCorrelatedColumnsInGroupBy(
+          invalidCols.map(_.name).mkString(","), expr.origin)
```

Comment:
@gengliangwang is working on the origin. Link up with him on whether you should leave it out here

Reply:
With https://github.com/apache/spark/pull/37841 merged, we can skip passing expr.origin here. @dtenedor Will you give a try?